gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/TopicInfo.java

```diff
@@ -98,15 +98,16 @@ public int hashCode() {
     return Objects.hash(name);
   }
 
+  final boolean baseEquals(TopicInfo topicInfo) {
+    return Objects.equals(name, topicInfo.name);
+  }
+
   @Override
   public boolean equals(Object obj) {
-    if (this == obj) {
-      return true;
-    }
-    if (obj == null || !obj.getClass().equals(this.getClass())) {
-      return false;
-    }
-    return Objects.equals(name, ((TopicInfo) obj).name);
+    return obj == this
+        || obj != null
```

Comment:
@jgeewax I understand your concerns, my inclination was to ask for additional parens, but I looked closer and at two referenced methods and am inclined to say Ok w/ this (and the prior one), unless you object.